Add GitHub Actions workflow to publish releases to PyPI on tag

traversaro opened the pull request:
Before doing the first tag after this PR is merged, it is necessary to add the PyPI token as a secret accessible to this repo under the name `PYPI_API_TOKEN`, see https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/ for more details.

traversaro committed:
Add GitHub Actions workflow to publish releases to PyPI on tag

fishbotics commented:
I think I added the secrets correctly. @traversaro: is there a way to test the action?

traversaro commented:
> I think I added the secrets correctly. @traversaro: is there a way to test the action?

In https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/ they suggest to also have a job that is publishing in TestPyPI, however after adding it in https://github.com/ami-iit/murdfpy/pull/2 I removed it in https://github.com/ami-iit/murdfpy/pull/5 due to some errors, after I checked the upload on regular PyPI were working on tag in https://github.com/ami-iit/murdfpy/actions/runs/4698910640 .

I guess in theory one can test on a fork and the TestPyPI instance, but in practice it may be just easier to start doing a tag, and if something goes wrong just upload the relase manually for that release.

fishbotics commented:
OK for now let's just ship it without the Test instance. We can fix that later if necessary.

traversaro commented:
> OK for now let's just ship it without the Test instance. We can fix that later if necessary.

Cool, thanks! Do you think it is ok but the version number in setup.py and tag 0.0.27? So we can both test the action, and I can package urchin in conda-forge, so that we can start migrating our projects from urdfpy to urchin, thanks!

fishbotics commented:
Updated!
